src/components/map/SplitView.js

```diff
@@ -7,104 +7,96 @@ import Layer from './layers/Layer';
-                        index={index}
-                        count={periods.length}
-                        layerId={id}
-                        onAdd={this.onMapAdd}
-                        onRemove={this.onMapRemove}
-                        setMapControls={this.setMapControls}
-                        isPlugin={isPlugin}
-                        isFullscreen={isFullscreen}
-                    >
-                        <Layer index={0} {...basemap} />
-                        <ThematicLayer
-                            index={1}
-                            period={period}
-                            feature={feature}
-                            {...layer}
-                            openContextMenu={openContextMenu}
-                        />
-                        <PeriodName period={period.name} />
-                    </MapItem>
-                ))}
-            </div>
-        );
-    }
+    useEffect(() => {
+        // From map plugin resize method
+        setShowFullscreen(isFullscreen);
+    }, [isFullscreen]);
 
-    // Called from map child
-    setMapControls = map => {
-        const controls = {};
+    const { id, periods = [] } = layer;
 
-        this.props.controls.forEach(control => {
-            map.addControl(control);
-            controls[control.type] = map.getControlContainer(control.type);
-        });
+    return (
+        <div
+            ref={containerRef}
+            className={cx('dhis2-map-split-view', styles.splitView)}
+        >
+            {periods.map((period, index) => (
+                <MapItem
+                    key={period.id}
+                    index={index}
+                    count={periods.length}
+                    layerId={id}
+                    setMapControls={setMap}
+                    isPlugin={isPlugin}
+                    isFullscreen={showFullscreen}
+                >
+                    <Layer index={0} {...basemap} />
+                    <ThematicLayer
+                        index={1}
+                        period={period}
+                        feature={feature}
+                        {...layer}
+                        openContextMenu={openContextMenu}
+                    />
+                    <PeriodName period={period.name} />
+                </MapItem>
+            ))}
+        </div>
+    );
+};
 
-        if (this.props.isPlugin) {
-            map.on('fullscreenchange', this.onFullScreenChange);
-        }
+SplitView.propTypes = {
+    isPlugin: PropTypes.bool,
+    isFullscreen: PropTypes.bool,
+    layer: PropTypes.object.isRequired,
+    basemap: PropTypes.object,
+    feature: PropTypes.object,
+    controls: PropTypes.array,
+    openContextMenu: PropTypes.func.isRequired,
+};
 
-        this.setState({ controls });
-    };
-
-    // From built-in fullscreen control
-    onFullScreenChange = ({ isFullscreen }) => this.setState({ isFullscreen });
-}
+SplitView.defaultProps = {
+    openContextMenu: () => {},
```

Comment:
Should this maybe be on the base Layer component instead? (also the "isRequired"  part of Line 95). I'm not an expert though on these things.